administrator/components/com_config/src/Model/ApplicationModel.php

```diff
@@ -585,11 +585,11 @@ public function save($data)
         }
 
         // Give a warning if the cache-folder can not be opened
-        if ($data['caching'] > 0 && $data['cache_handler'] == 'file' && @opendir($path) == false) {
+        if ($data['caching'] > 0 && $data['cache_handler'] == 'file' && !@opendir($path)) {
             $error = true;
 
             // If a custom path is in use, try using the system default instead of disabling cache
-            if ($path !== JPATH_CACHE && @opendir(JPATH_CACHE) != false) {
+            if ($path !== JPATH_CACHE && @opendir(JPATH_CACHE)) {
```

Comment:
Same as above, better use strict comparison with false
```suggestion
            if ($path !== JPATH_CACHE && @opendir(JPATH_CACHE) !== false) {
```